pkg/fetcher/message_fetcher.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2025 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fetcher
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/th2-net/th2-common-go/pkg/grpc"
+	common "github.com/th2-net/th2-grpc-common-go"
+	lwdp "github.com/th2-net/th2-grpc-lw-data-provider-go"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	LwdpBase64Format     = "BASE_64"
+	LwdpJsonParsedFormat = "JSON_PARSED"
+	lwdpService          = "DataProviderService"
+)
+
+func GetLastGroupedMessage(router grpc.Router, book string, group string, alias string, direction common.Direction, format string, deadlineMs uint32) (*lwdp.MessageGroupResponse, error) {
```

Comment:
Just a thought: maybe it would be better if we create a struct with a method to fetch the data instead of a function
```golang
package fether

type Fether struct {
}

func New(router grpc.Router) (Fetcher, error) {}

func (f *Fether) GetLastGroupedMessage(ctx context.Context, book string, group string, alias string, direction common.Direction, format string) (*lwdp.MessageGroupResponse, error) {}
```